openstack/messaging/v2/claims/requests.go

```diff
@@ -0,0 +1,73 @@
+package claims
+
+import (
+	"net/http"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder Builder.
+type CreateOptsBuilder interface {
+	ToClaimCreateRequest() (map[string]interface{}, string, error)
+}
+
+// CreateOpts params to be used with Create.
+type CreateOpts struct {
+	// The name of the queue to claim messages from.
+	QueueName string `json:"queue_name" required:"true"`
```

Comment:
The original time we did this was a very unique situation: an attribute of the resource was part of the URL and it felt really weird having it in the function signature.

An equivalent here would be if `Grace` was part of the URL rather in the request body.

Since the Queue Name isn't part of the `Claim` itself, it's a relational attribute, it probably should go in the `Create` function signature:

```
claim, err := claims.Create(client, queueName, opts).Extract()
```

Given `messages.Create` has `queueName` in the URL, too, we should follow that for consistency: https://github.com/gophercloud/gophercloud/blob/master/openstack/messaging/v2/messages/requests.go#L38

_technically_ `Claims` could even be bundled in the `queues` package. This happens occasionally - for example LBaaS has Pools and Members bundled because they share the same URL path. That's always felt a little weird to me, but if it's something you'd prefer to do, go for it.

Reply:
Personally I like having it separate from queues. Mainly because doing it the other way is a little confusing.